tests/test_protocol.py

```diff
@@ -1441,3 +1441,62 @@ class TestServer(ServerSlice):
     response = await client.test_method()
     assert response.code == 200
     assert response.result == zip_content
+
+
+@pytest.fixture
+async def options_server():
+    @protocol.typedmethod(path="/test", operation="OPTIONS", client_types=["api"])
+    def test_method() -> ReturnValue[bytes]:  # NOQA
+        pass
+
+    class TestServer(ServerSlice):
+        @protocol.handle(test_method)
+        async def test_methodY(self) -> ReturnValue[str]:  # NOQA
+            return ReturnValue(response="content")
+
+    return TestServer(name="testserver")
+
+
+@pytest.fixture
+def options_request(unused_tcp_port):
```

Comment:
Why don't we use protocol.Client()?

Reply:
Because the Result object it returns doesn't include the headers